src/modules/cxxpiper.cpp

```diff
@@ -0,0 +1,1223 @@
+	optional<float> sentenceSilenceSeconds;
+	// Path to espeak-ng data directory (default is next to piper executable)
+	optional<filesystem::path> eSpeakDataPath;
+	// Path to libtashkeel ort model
+	optional<filesystem::path> tashkeelModelPath;
+	bool jsonInput = false;
+	// Seconds of extra silence to insert after a single phoneme
+	optional<std::map<piper::Phoneme, float>> phonemeSilenceSeconds;
+	// true to use CUDA execution provider
+	bool useCuda = false;
+    };
+    static RunConfig runConfig;
+    static piper::Voice voice;
+    static SPDVoice **cxxpiper_voice_list = NULL;
+
+    static int cxxpiper_alloc_cbuf();
+    static SPDVoice **cxxpiper_allocate_voice_list();
+    static void cxxpiper_free_voice_list();
+    static void cxxpiper_handle_text(const char *);
+    static void cxxpiper_handle_sound_icon(const char *);
+    static void cxxpiper_set_language(char *);
+    static void cxxpiper_set_synthesis_voice(char *voice_name);
+    static void cxxpiper_set_voice_type(SPDVoiceType);
+  static int cxxpiper_voice_name_to_speaker_id(const char *);
+
+    MOD_OPTION_1_INT(UseCUDA)
+    MOD_OPTION_1_STR(ModelPath)
+    MOD_OPTION_1_STR(ConfigPath)
+    MOD_OPTION_1_STR(SoundIconFolder)
+    MOD_OPTION_1_INT(SoundIconVolume)
+    // SIC: see below regarding NONE and ALL
+    MOD_OPTION_1_STR(PunctSome)
+    MOD_OPTION_1_STR(PunctMost)
+    MOD_OPTION_1_STR(ESpeakNGDataDirPath)
+
+    int module_load(void)
+    {
+	INIT_SETTINGS_TABLES();
+	REGISTER_DEBUG();
+	MOD_OPTION_1_INT_REG(UseCUDA, 0);
+	MOD_OPTION_1_STR_REG(ModelPath, "");
+	MOD_OPTION_1_STR_REG(ConfigPath, "");
+	MOD_OPTION_1_STR_REG(SoundIconFolder, "/usr/share/sounds/sound-icons/");
+	MOD_OPTION_1_INT_REG(SoundIconVolume, 0);
+	// NB:  NONE and ALL are constant so no config.  NONE means omit as much as possible
+	// while ALL  means omit as little as possible.
+	// NB: Don't provide defaults, since empty string means fallback to SD's definitions.
+	MOD_OPTION_1_STR_REG(PunctSome, "");
+	MOD_OPTION_1_STR_REG(PunctMost, "");
+	MOD_OPTION_1_STR_REG(ESpeakNGDataDirPath, "/usr/share/espeak-ng-data/");
+	module_register_available_voices();
+	module_register_settings_voices();
+	return 0;
+    }
+
+    int module_init(char **status_info)
+    {
+	module_audio_set_server();
+	module_audio_init(status_info);
+	runConfig.useCuda = UseCUDA;
+	runConfig.modelPath = filesystem::path(ModelPath);
+	runConfig.modelConfigPath = filesystem::path(ConfigPath);
+	runConfig.eSpeakDataPath = filesystem::path(ESpeakNGDataDirPath);
+	runConfig.outputPath = nullopt;
+	char *default_voice = module_getdefaultvoice();
+	runConfig.defaultVoiceName = default_voice;
+	runConfig.speakerId = cxxpiper_voice_name_to_speaker_id(runConfig.defaultVoiceName);
+	DBG("Default Voice is %s", runConfig.defaultVoiceName);
+	cxxpiper::loadVoice(piperConfig, runConfig.modelPath.string(), runConfig.modelConfigPath.string(),
+				voice, runConfig.speakerId, runConfig.useCuda);
```

Comment:
We really need to catch errors here, and return -1 on failure, to make sure that the voice doesn't show up in orca's list and users accidently enabling a voice that actually doesn't work and then they're doomed

Reply:
This is really a must